networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/AttestationGossipManager.java

```diff
@@ -63,25 +69,41 @@ public void onNewAttestation(final ValidateableAttestation validateableAttestati
               attestationPublishSuccessCounter.inc();
             },
             error -> {
-              if (Throwables.getRootCause(error) instanceof MessageAlreadySeenException) {
-                LOG.debug(
-                    "Failed to publish attestation for slot {} because the message has already been seen",
-                    attestation.getData().getSlot());
-              } else if (Throwables.getRootCause(error)
-                  instanceof NoPeersForOutboundMessageException) {
-                LOG.warn(
-                    "Failed to publish attestation for slot {} because no peers were available on the required gossip topic",
-                    attestation.getData().getSlot());
-              } else {
-                LOG.error(
-                    "Failed to publish attestation for slot {}",
-                    attestation.getData().getSlot(),
-                    error);
-              }
+              logWithSuppression(error, attestation);
               attestationPublishFailureCounter.inc();
             });
   }
 
+  synchronized void logWithSuppression(final Throwable error, final Attestation attestation) {
+    final AttestationData attestationData = attestation.getData();
+    final Throwable rootCause = Throwables.getRootCause(error);
+    if (attestationData.getSlot().equals(lastErroredSlot)
+        && attestationData.getIndex().equals(lastErroredCommitteeIndex)
+        && rootCause.equals(lastRootCause)) {
+      return;
+    }
+    lastErroredSlot = attestationData.getSlot();
+    lastErroredCommitteeIndex = attestationData.getIndex();
```

Comment:
I'm slightly worried that the attestations won't always be published in order of committee index so we may wind up logging multiple warnings still.  I'd be very tempted to omit the committee index and just print the warning once per slot.  It's kind of useful to know the committee index but not particularly because you can't really do much.

Reply:
Good point